feat: adding the ability to pass a regex to filter tickets

wovalle opened the pull request:
Hey, nice project!

In my company we have many jira projects and instead of passing a fixed array of prefixes, we needed to pass a regex and get the ticket name on the fly. 

I added some documentation around it.

Thanks!

GabrielCastro commented:
Thanks for the contribution

GabrielCastro commented on lib/success.ts:
```diff
@@ -6,10 +6,16 @@ import { GenerateNotesContext, PluginConfig } from './types';
+  let patterns: RegExp[] = [];
+
+  if (config.ticketRegex) {
+    patterns = [new RegExp(escapeRegExp(config.ticketRegex), 'giu')];
```
we should remove the `escapeRegExp` call here otherwise it won't match

wovalle commented:
Hey! Sorry it took me so long to reply, I was on vacations!

I updated the function and added a couple of tests to verify that everything work correctly :)

wovalle commented:
cc @guigonc

GabrielCastro commented:
Thanks for the change including setting up tests 💯

GabrielCastro commented:
:tada: This PR is included in version 0.3.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/semantic-release-jira-releases)
- [GitHub release](https://github.com/UpHabit/semantic-release-jira-releases/releases/tag/v0.3.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
